ee-feature-pack/ee-8-api/src/main/resources/modules/system/layers/base/javax/persistence/api/main/module.xml

```diff
@@ -22,13 +22,17 @@
   ~ 02110-1301 USA, or see the FSF site: http://www.fsf.org.
   -->
 
-<module xmlns="urn:jboss:module:1.7" name="javax.persistence.api">
-    <dependencies>
-        <!-- PersistenceUnitInfo needs javax.sql.DataSource -->
-        <module name="javax.api" export="true"/>
-    </dependencies>
+<module xmlns="urn:jboss:module:1.9" name="javax.persistence.api">
 
     <resources>
         <artifact name="${jakarta.persistence:jakarta.persistence-api}"/>
     </resources>
+
+    <dependencies>
+        <module name="java.instrument"/>
```

Comment:
The ee-9 variant of this has export=true:

```
        <!-- java.instrument export due to jdeps result -->
        <module name="java.instrument" export="true"/>
```

I think that's needed here as ClassTransformer throws IllegalClassFormatException in JPA 2.2 as well:

https://github.com/eclipse-ee4j/jpa-api/blob/2.2-2.2.3-RELEASE/api/src/main/java/javax/persistence/spi/ClassTransformer.java#L61

Reply:
@bstansberry 
Thanks for your feedback. I have fixed it.